include/RAJA/exec-cuda/forall_cuda.hxx

```diff
@@ -175,18 +204,26 @@ template <size_t BLOCK_SIZE, typename LOOP_BODY>
 RAJA_INLINE void forall(cuda_exec<BLOCK_SIZE>,
                         Index_type begin,
                         Index_type end,
-                        LOOP_BODY loop_body)
+                        LOOP_BODY&& loop_body)
 {
+  beforeCudaKernelLaunch();
```

Comment:
Have we measured the overhead of these before/after calls for kernels without reductions?